### PR TITLE
HDDS-5286. Document SCM HA classes

### PR DESCRIPTION
**Pull request by sky76093016**
## What changes were proposed in this pull request?

Add class-level javadoc for SCMDBCheckpointProvider and SCMHANodeDetails.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/projects/HDDS/issues/HDDS-5286

## How was this patch tested?

No test.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMDBCheckpointProvider.java [hunk 1]
```diff
@@ -31,6 +31,10 @@
 
 
 // TODO: define a generic interface for this
+
+/**
+ * Checkpoint write sream and exception handling.
```
Comment (xiaoyuyao) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMDBCheckpointProvider.java:36]: sream=> stream?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHANodeDetails.java [hunk 2]
```diff
@@ -60,6 +60,9 @@
 import static org.apache.hadoop.hdds.scm.ScmConfigKeys.OZONE_SCM_SECURITY_SERVICE_PORT_KEY;
 import static org.apache.hadoop.hdds.scm.ScmConfigKeys.OZONE_SCM_SERVICE_IDS_KEY;
 
+/**
+ * Construct SCM HA node details.
```
Comment (xiaoyuyao) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHANodeDetails.java:64]: For class doc, can we remove "Construct"

**Review by xiaoyuyao — COMMENTED** [review]
LGTM, minor comments inline

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMDBCheckpointProvider.java (hunk 1 above)
Comment (sky76093016) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMDBCheckpointProvider.java:36]: Solve it.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHANodeDetails.java (hunk 2 above)
Comment (sky76093016) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHANodeDetails.java:64]: Solve it.

**Comment by xiaoyuyao** [comment]
LGTM, +1.

**Pull request merged**